borgmatic/borg/create.py

```diff
@@ -291,6 +291,7 @@ def collect_special_file_paths(
         capture_stderr=True,
         working_directory=working_directory,
         extra_environment=borg_environment,
+        raise_on_exit_code_one=False,
```

Comment:
Adding this parameter to `execute_command_and_capture_output()` is a clever way to change the error behavior of just `collect_special_file_paths()` without affecting other code, but the name of the parameter "leaks" implementation details about what's considered an error. Maybe it could be named something more generic like `raise_on_exit_code_warning` or `treat_exit_code_warning_as_error` or `error_on_warning` or whatever.

I also wonder a little whether the parameter is needed at all. Maybe this same type of bug exists in some of the other callers of `execute_command_and_capture_output()`.... Having wondered that aloud, I don't think you need to expand scope to encompass the other callers right now, but it's maybe something to think about or look out for in the future.

Reply:
borgmatic has 100%-ish test coverage, but it's mostly unit tests instead of integration tests. So if some callers in one unit are implicitly relying on Borg warnings being treated as errors in another unit, that won't necessarily be captured in any existing test.

Having said that, I went ahead and manually reviewed all callers of `execute_command_and_capture_output()`, and I didn't immediately see anything that would be problematic. And just thinking this through, if some existing code calls Borg and now doesn't receive an exception when Borg returns exit code 1, that should hopefully not hurt anything and is arguably more correct.

The one case I did test manually was when `borgmatic init` probes for repository existence with `borg rinfo`/`borg info`. And I did confirm that when the repository doesn't exist, Borg helpfully returns exit code 2 instead of 1.

So my recommendation is to proceed with your approach!